### PR TITLE
Refactor rest test

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #

- [x] Tests
- ~~Layman Test Client~~
- [x] Changlelog~~
- ~~Documentation~~

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: src/layman/layer/rest_workspace_test.py [hunk 1]
```diff
@@ -63,7 +63,7 @@
 }
 """
 
-num_layers_before_test = 0  # pylint: disable=invalid-name
+num_of_publications = PublicationCounter()
```
Comment (jirik) [inline src/layman/layer/rest_workspace_test.py]: `num_of_publications` -> `PUBLICATION_COUNTER` (not sure, why pylint accepts it in lowercase)

File: src/layman/map/rest_workspace_test.py [hunk 2]
```diff
@@ -38,7 +38,7 @@
 
 METADATA_PROPERTIES_EQUAL = METADATA_PROPERTIES
 
-num_maps_before_test = 0  # pylint: disable=invalid-name
+num_of_publications = PublicationCounter()
```
Comment (jirik) [inline src/layman/map/rest_workspace_test.py]: `num_of_publications` -> `PUBLICATION_COUNTER` (not sure, why pylint accepts it in lowercase)

File: src/layman/layer/qgis/qgis_test.py [hunk 3]
```diff
@@ -29,6 +29,9 @@ def test_qgis_rest():
         assert wms.get_layer_info(workspace, layer) == {'name': layer,
                                                         'style': {'type': 'qml',
                                                                   'url': url},
+                                                        'wms': {
```
Comment (jirik) [inline src/layman/layer/qgis/qgis_test.py:32]: `'wms'` -> `'_wms'` (it should be internal)

File: src/layman/layer/qgis/wms.py [hunk 4]
```diff
@@ -27,6 +27,9 @@ def get_layer_info(workspace, layername):
                 'url': url,
                 'type': 'qml',
             },
+            'wms': {
```
Comment (jirik) [inline src/layman/layer/qgis/wms.py:30]: `'wms'` -> `'_wms'` (it should be internal)

File: src/layman/layer/geoserver/__init__.py [hunk 5]
```diff
@@ -160,10 +159,11 @@ def publish_layer_from_db(workspace, layername, description, title, access_right
 def publish_layer_from_qgis(workspace, layer, description, title, access_rights, geoserver_workspace=None):
     geoserver_workspace = geoserver_workspace or workspace
     store_name = wms.get_qgis_store_name(layer)
+    layer_capabilities_url = layman_util.get_publication_info(workspace, LAYER_TYPE, layer, context={'keys': ['wms']})['wms']['qgis_capabilities_url']
```
Comment (jirik) [inline src/layman/layer/geoserver/__init__.py:162]: `['wms']['qgis_capabilities_url']` -> `['_wms']['qgis_capabilities_url']`

File: src/layman/map/client_test.py [hunk 6]
```diff
@@ -54,9 +48,7 @@ def chrome():
 @pytest.mark.test_client
 @pytest.mark.usefixtures('ensure_layman', 'client')
 def test_post_no_file(chrome):
-    check_redis_consistency(expected_publ_num_by_type={
-        f'{MAP_TYPE}': num_maps_before_test
-    })
+    check_redis_consistency()
```
Comment (jirik) [inline src/layman/map/client_test.py:51]: I think
```
    check_redis_consistency(expected_publ_num_by_type={
        f'{MAP_TYPE}': 0
    })
```
is closer to original.

File: src/layman/map/client_test.py [hunk 7]
```diff
@@ -110,6 +102,4 @@ def test_post_no_file(chrome):
                                            ) and entry['message'].endswith(
             'Failed to load resource: the server responded with a status of 400 (BAD REQUEST)')
 
-    check_redis_consistency(expected_publ_num_by_type={
-        f'{MAP_TYPE}': num_maps_before_test
-    })
+    check_redis_consistency()
```
Comment (jirik) [inline src/layman/map/client_test.py:105]: I think
```
    check_redis_consistency(expected_publ_num_by_type={
        f'{MAP_TYPE}': 0
    })
```
is closer to original.

File: test/prime_db_schema_client.py [hunk 8]
```diff
@@ -1,8 +1,8 @@
 import uuid
-from layman import settings, app
+from layman import settings, app, util as layman_util
 from layman.common.prime_db_schema import publications, workspaces, users
 
-latest_oauth2_sub = 0  # pylint: disable=invalid-name
+latest_oauth2_sub = layman_util.SimpleCounter()
```
Comment (jirik) [inline test/prime_db_schema_client.py:5]: `latest_oauth2_sub` -> `OAUTH2_SUB_COUNTER`

File: test/process.py [hunk 9]
```diff
@@ -27,7 +28,7 @@
 
 LAYMAN_SETTING = None
 LAYMAN_DEFAULT_SETTINGS = AUTHN_SETTINGS
-LAYMAN_START_COUNT = 0
+layman_start_count = layman_util.SimpleCounter()
```
Comment (jirik) [inline test/process.py:31]: `layman_start_count` -> `LAYMAN_START_COUNTER`

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/layer/rest_workspace_test.py (hunk 1 above)
Comment (index-git) [reply src/layman/layer/rest_workspace_test.py]: I think, that uppercase is for constants and this is not constant. It is a global variable, so I used lowercase.

**Review by jirik — APPROVED** [review]

**Pull request merged**